kb/2023-08-21/compatible_with_ip_pool_new_feature.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Upgrade Guest Kubernetes Clusters to be Compatible with Harvester IP Pools
+description: Explain how to keep load balancer IP during upgrading guest cluster
+slug: upgrading_guest_clusters_with_harvester_ip_pool_compatibility
+authors:
+- name: Canwu Yao
+  title: Software Engineer
+  url: https://github.com/yaocw2020
+  image_url: https://avatars.githubusercontent.com/u/7421463?s=400&v=4
+tags: [harvester, load balancer, cloud provider, ip pool, upgrade] 
+hide_table_of_contents: false
+---
+
+As **Harvester v1.2.0** is released, a new Harvester cloud provider version **0.2.2** is integrated into the RKE2 `v1.24.15+rke2r1`, `v1.25.11+rke2r1`,  `v1.26.6+rke2r1`, `v1.27.3+rke2r1`, and newer versions.
+
+With Harvester v1.2.0, the new Harvester cloud provider offers enhanced load balancing capabilities for guest Kubernetes services. Introducing the Harvester IP Pool feature, a built-in IP address management (IPAM) solution for the Harvester load balancer, allows users to define an IP pool specific to a particular guest cluster by specifying the guest cluster name. For example, you can create an IP pool exclusively for the guest cluster named cluster2:
+
+![image](ippoolforcluster2.png)
+
+However, the feature is not automatically compatible with existing guest Kubernetes clusters after upgrading, as they do not pass the correct cluster name to the Harvester cloud provider (refer to [issue 4232](https://github.com/harvester/harvester/issues/4232)). Users can manually upgrade the Harvester cloud provider using Helm as a workaround and provide the correct cluster name after upgrading. However, this would result in a change in the load balancer IPs. This article outlines a workaround that allows users to leverage the new IP pool feature while keeping the load balancer IPs unchanged.
```

Comment:
```suggestion
However, after upgrading, the feature is not automatically compatible with existing guest Kubernetes clusters, as they do not pass the correct cluster name to the Harvester cloud provider. Refer to [issue 4232](https://github.com/harvester/harvester/issues/4232) for more details. Users can manually upgrade the Harvester cloud provider using Helm as a workaround and provide the correct cluster name after upgrading. However, this would result in a change in the load balancer IPs. 

This article outlines a workaround that allows you to leverage the new IP pool feature while keeping the load balancer IPs unchanged.
```